hazelcast/src/main/java/com/hazelcast/spi/impl/eventservice/impl/EventServiceImpl.java

```diff
@@ -441,16 +492,10 @@ public void executeEventCallback(Runnable callback) {
 
     @Override
     public void handle(Packet packet) {
-        try {
-            eventExecutor.execute(new RemoteEventProcessor(this, packet));
-        } catch (RejectedExecutionException e) {
-            rejectedCount.inc();
-
-            if (eventExecutor.isLive()) {
-                Connection conn = packet.getConn();
-                String endpoint = conn.getEndPoint() != null ? conn.getEndPoint().toString() : conn.toString();
-                logFailure("EventQueue overloaded! Failed to process event packet sent from: %s", endpoint);
-            }
+        EventEnvelope eventEnvelope = (EventEnvelope) nodeEngine.toObject(packet);
```

Comment:
not that it's terribly important, but no HotSpot version I know of uses `System.currentTimeMillis()` to generate identity hashcode. 

See [this](http://hg.openjdk.java.net/jdk8u/jdk8u/hotspot/file/981cb5ddafc8/src/share/vm/runtime/synchronizer.cpp#l557) for available strategies. The last strategy is used by default (in recent versions of HotSpot). The strategy no. 2 could be a lot of fun. `-XX:hashCode=2` :)
`</smartass-mode>`